fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckPolicy.java

```diff
@@ -40,22 +41,34 @@ public class CheckPolicy implements AnalysisRuleFactory {
     @Override
     public List<Rule> buildRules() {
         return ImmutableList.of(
-            RuleType.CHECK_ROW_POLICY.build(
-                logicalCheckPolicy(any().when(child -> !(child instanceof UnboundRelation))).thenApply(ctx -> {
-                    LogicalCheckPolicy<Plan> checkPolicy = ctx.root;
-                    Plan child = checkPolicy.child();
-                    if (!(child instanceof LogicalRelation)
-                            || ctx.connectContext.getSessionVariable().isPlayNereidsDump()) {
-                        return child;
-                    }
-                    LogicalRelation relation = (LogicalRelation) child;
-                    Optional<Expression> filter = checkPolicy.getFilter(relation, ctx.connectContext);
-                    if (!filter.isPresent()) {
-                        return relation;
-                    }
-                    return new LogicalFilter(ExpressionUtils.extractConjunctionToSet(filter.get()), relation);
-                })
-            )
+                RuleType.CHECK_ROW_POLICY.build(
+                        logicalCheckPolicy(any().when(child -> !(child instanceof UnboundRelation))).thenApply(ctx -> {
+                            LogicalCheckPolicy<Plan> checkPolicy = ctx.root;
+                            LogicalFilter<Plan> upperFilter = null;
+
+                            Plan child = checkPolicy.child();
+                            if (child instanceof LogicalFilter && child.bound() && child
```

Comment:
why check `child bound`

Reply:
Because the unique table will automatically include a filter condition